DataSource/Util/ChoicesBuilder.php

```diff
@@ -91,15 +122,36 @@ private function getChoicesFromRepository(EntityRepository $repository, $propert
     }
 
     /**
-     * @deprecated Repository should be specified by it's name, not as an actual object.
-     *
      * @param EntityRepository $repository
+     * @param string           $method
      *
-     * @return EntityRepository
+     * @return mixed
+     *
+     * @throws \Exception
      */
-    private function getSpecifiedRepository(EntityRepository $repository)
+    private function getChoicesFromRepositoryWithMethod(EntityRepository $repository, $method)
     {
-        trigger_error('Repository should be specified by it\'s name, not as an actual object.', E_USER_DEPRECATED);
-        return $repository;
+        if (!method_exists($repository, $method)) {
+            throw new \Exception("Specified repository does not have '$method' method");
+        }
+
+        return $repository->$method();
```

Comment:
should we sanitise what this method returns? What if it returns something that cannot be handled as choices